plugin/plugin.go

```diff
@@ -213,13 +232,55 @@ func (p *OrmPlugin) parseBasicFields(msg *generator.Descriptor) {
 	for _, field := range getMessageOptions(msg).GetInclude() {
 		fieldName := generator.CamelCase(field.GetName())
 		if _, ok := ormable.Fields[fieldName]; !ok {
-			ormable.Fields[fieldName] = &Field{Type: field.GetType(), GormFieldOptions: &gorm.GormFieldOptions{Tag: field.GetTag()}}
+			p.addIncludedField(ormable, field)
 		} else {
 			p.Fail("Cannot include", fieldName, "field into", ormable.Name, "as it aready exists there.")
 		}
 	}
 }
 
+func (p *OrmPlugin) addIncludedField(ormable *OrmableType, field *gorm.ExtraField) {
+	fieldName := generator.CamelCase(field.GetName())
+	// Handle non-imported types
+	if _, ok := builtinTypes[strings.TrimPrefix(field.GetType(), "*")]; ok {
+		// basic type, 100% okay
+	} else if field.GetType() == "time.Time" {
+		p.GetFileImports().usingTime = true
+	} else if field.GetType() == "uuid.UUID" || field.GetType() == "*uuid.UUID" {
+		p.GetFileImports().usingUUID = true
+	} else if (field.GetType() == "postgres.Jsonb" || field.GetType() == "pq.Jsonb" ||
+		field.GetType() == "gormpq.Jsonb") && p.dbEngine == ENGINE_POSTGRES {
+		p.GetFileImports().usingJSON = true
+		fType := "gormpq.Jsonb"
+		field.Type = &fType
+	} else {
+		// Not a built-in or super-special type
+		parts := strings.Split(field.GetType(), ".")
+		// Type format is something other than package.type
+		if len(parts) != 2 {
+			p.Fail(
+				fmt.Sprintf(`Included field %q of type %q does not match pattern package.type`,
+					field.GetName(), field.GetType()))
+		}
+		if field.GetPackage() != "" {
+			if v, ok := p.GetFileImports().githubImports[parts[0]]; ok && v != field.GetPackage() {
+				p.Fail(
+					fmt.Sprintf(`Included field %q of type %q reuses package prefix %q, but has different package %q, was %q`,
+						field.GetName(), field.GetType(), parts[0], field.GetPackage(), v))
+			}
+			p.GetFileImports().githubImports[parts[0]] = field.GetPackage()
+		} else if v, ok := p.GetFileImports().githubImports[parts[0]]; !ok {
+			p.Fail(fmt.Sprintf(`Included field %q of type %q not a recognized special type, and no package specified`, field.GetName(), field.GetType()))
+		} else {
+			// If this package prefix has already been used with a package definition,
+			// let's not give up, but let them know and hope for the best
+			log.Printf(`Included field %q of type %q reuses package prefix %q, assumed to also be from package %q`,
```

Comment:
What is the need of specifying alias along with the type name? It even complicates things since you need to validate if the user did a mistake by duplicating aliases.